From a2fdb0e96b2d1ef3d3e435f9c5d57d2d5f498790 Mon Sep 17 00:00:00 2001
From: Igor Patrik <igorp@marvell.com>
Date: Tue, 9 Jul 2013 07:41:36 +0200
Subject: [PATCH 0772/1825] axp: Added support for AMC

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit fb2eab386b934cf0b2db4fc0e2818d22bd0455f2

	Added AMC case in mach-types, Kconfig and core.c
	In defconfig changed number of ports to 4:
	eth0 not functional in linux, so working only egiga0/eth1, TODO.

Change-Id: I48de7eb42c2d581b4568e8476c126ab757466ae3
Signed-off-by: Igor Patrik <igorp@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/2456
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/configs/armada_xp_bc2_v7smp_defconfig |    2 +-
 arch/arm/mach-armadaxp/Kconfig                 |    9 +++++++++
 arch/arm/mach-armadaxp/core.c                  |   19 +++++++++++++++++++
 arch/arm/tools/mach-types                      |    1 +
 4 files changed, 30 insertions(+), 1 deletions(-)

diff --git a/arch/arm/configs/armada_xp_bc2_v7smp_defconfig b/arch/arm/configs/armada_xp_bc2_v7smp_defconfig
index 031689f..97abe3f 100644
--- a/arch/arm/configs/armada_xp_bc2_v7smp_defconfig
+++ b/arch/arm/configs/armada_xp_bc2_v7smp_defconfig
@@ -20,7 +20,7 @@ CONFIG_PARTITION_ADVANCED=y
 CONFIG_EFI_PARTITION=y
 CONFIG_ARCH_ARMADA_XP=y
 # CONFIG_MV_INCLUDE_SWITCH is not set
-CONFIG_MV_ETH_PORTS_NUM=2
+CONFIG_MV_ETH_PORTS_NUM=4
 CONFIG_MV_ETH_RXQ=8
 CONFIG_MV_ETH_TXQ=8
 CONFIG_NET_SKB_HEADROOM=96
diff --git a/arch/arm/mach-armadaxp/Kconfig b/arch/arm/mach-armadaxp/Kconfig
index 2a8f51c..70e270c 100644
--- a/arch/arm/mach-armadaxp/Kconfig
+++ b/arch/arm/mach-armadaxp/Kconfig
@@ -81,6 +81,15 @@ config MACH_ARMADA_XP_RD_NAS
 	default y
 	help
 
+config MACH_ARMADA_XP_AMC
+	bool "Marvell Armada XP AMC Board"
+	default y
+	help
+         This board is Marvell AMC card
+	 having 2 network ports
+	 connected to phys 0 and 3
+	 This board used primary as a management of switch boards
+
 config MACH_ARMADA_XP_FPGA
 	bool "Marvell Armada XP FPGA Board"	
 	depends on !MACH_ARMADA_XP_DB && !MACH_ARMADA_XP_RDSRV
diff --git a/arch/arm/mach-armadaxp/core.c b/arch/arm/mach-armadaxp/core.c
index 87a5cc2..c873ff0 100644
--- a/arch/arm/mach-armadaxp/core.c
+++ b/arch/arm/mach-armadaxp/core.c
@@ -1901,6 +1901,25 @@ MACHINE_START(ARMADA_XP_GP, "Marvell Armada XP GP Board")
 #endif
 MACHINE_END
 
+MACHINE_START(ARMADA_XP_AMC, "Marvell Armada XP AMC Board")
+	/* MAINTAINER("MARVELL") */
+	.atag_offset	= 0x00000100,
+	.map_io		= axp_map_io,
+	.init_irq	= axp_init_irq,
+	.timer		= &axp_timer,
+	.init_machine	= axp_gp_init,
+#ifdef CONFIG_FB_DOVE
+	/* reserve memory for LCD */
+	.fixup		= axp_tag_fixup_mem32,
+#endif /* CONFIG_FB_DOVE */
+#ifdef CONFIG_SUSPEND
+	.reserve	= reserve_training_mem,
+#endif /* CONFIG_SUSPEND */
+#ifdef CONFIG_MV_AMP_ENABLE
+	.reserve	= axp_reserve_regs_space,
+#endif
+MACHINE_END
+
 /*****************************************************************************
  * RD NAS BOARD
  ****************************************************************************/
diff --git a/arch/arm/tools/mach-types b/arch/arm/tools/mach-types
index bde313e..329a027 100644
--- a/arch/arm/tools/mach-types
+++ b/arch/arm/tools/mach-types
@@ -520,6 +520,7 @@ armada_xp_gp		MACH_ARMADA_XP_GP	ARMADA_XP_GP		3035
 armada_xp_db		MACH_ARMADA_XP_DB	ARMADA_XP_DB		3036
 armada_xp_rdsrv		MACH_ARMADA_XP_RDSRV	ARMADA_XP_RDSRV		3037
 armada_370		MACH_ARMADA_370		ARMADA_370		3038
+armada_xp_amc		MACH_ARMADA_XP_AMC		ARMADA_XP_AMC		3039
 armada_xp_rd_nas	MACH_ARMADA_XP_RD_NAS	ARMADA_XP_RD_NAS	3500
 tx28			MACH_TX28		TX28			3043
 pcontrol_g20		MACH_PCONTROL_G20	PCONTROL_G20		3062
-- 
1.7.5.4

